lib/backend/model/rule.go

```diff
@@ -39,16 +39,20 @@ type Rule struct {
 	NotICMPType *int `json:"!icmp_type,omitempty" validate:"omitempty,gte=0,lt=255"`
 	NotICMPCode *int `json:"!icmp_code,omitempty" validate:"omitempty,gte=0,lte=255"`
 
-	SrcTag      string             `json:"src_tag,omitempty" validate:"omitempty,tag"`
-	SrcNet      *net.IPNet         `json:"src_net,omitempty" validate:"omitempty"`
-	SrcNets     []*net.IPNet       `json:"src_nets,omitempty" validate:"omitempty"`
-	SrcSelector string             `json:"src_selector,omitempty" validate:"omitempty,selector"`
-	SrcPorts    []numorstring.Port `json:"src_ports,omitempty" validate:"omitempty,dive"`
-	DstTag      string             `json:"dst_tag,omitempty" validate:"omitempty,tag"`
-	DstSelector string             `json:"dst_selector,omitempty" validate:"omitempty,selector"`
-	DstNet      *net.IPNet         `json:"dst_net,omitempty" validate:"omitempty"`
-	DstNets     []*net.IPNet       `json:"dst_nets,omitempty" validate:"omitempty"`
-	DstPorts    []numorstring.Port `json:"dst_ports,omitempty" validate:"omitempty,dive"`
+	SrcTag        string             `json:"src_tag,omitempty" validate:"omitempty,tag"`
+	SrcNet        *net.IPNet         `json:"src_net,omitempty" validate:"omitempty"`
+	SrcNets       []*net.IPNet       `json:"src_nets,omitempty" validate:"omitempty"`
+	SrcSelector   string             `json:"src_selector,omitempty" validate:"omitempty,selector"`
+	SrcPorts      []numorstring.Port `json:"src_ports,omitempty" validate:"omitempty,dive"`
+	SrcSANames    string             `json:"src_sa_name,omitempty" validate:"omitempty"`
+	SrcSASelector string             `json:"src_sa_selector,omitempty" validate:"omitempty"`
+	DstTag        string             `json:"dst_tag,omitempty" validate:"omitempty,tag"`
+	DstSelector   string             `json:"dst_selector,omitempty" validate:"omitempty,selector"`
+	DstNet        *net.IPNet         `json:"dst_net,omitempty" validate:"omitempty"`
+	DstNets       []*net.IPNet       `json:"dst_nets,omitempty" validate:"omitempty"`
+	DstPorts      []numorstring.Port `json:"dst_ports,omitempty" validate:"omitempty,dive"`
+	DstSANames    string             `json:"dst_sa_name,omitempty" validate:"omitempty"`
```

Comment:
Are there Felix changes for this too?  Just wondered why we needed to change the Felix model for this.

Reply:
There are no felix changes. tbh i was not sure if this is needed or not. i noticed we don't do it for namespaceSelector. 
Is it that all the sa based selector are going to get munged into the selector and therefore we don't need these?